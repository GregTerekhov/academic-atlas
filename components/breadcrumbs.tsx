export default function Breadcrumbs() {
  return (
    <>
      <p>Breadcrumbs</p>
    </>
  );
}
