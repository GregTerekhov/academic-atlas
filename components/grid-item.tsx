export default function GridItem() {
  return (
    <>
      <p>GridItem</p>
    </>
  );
}
