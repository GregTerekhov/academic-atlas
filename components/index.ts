export { default as AboutService } from './home/about';
export { default as Benefits } from './home/benefits';
export { default as Breadcrumbs } from './layout/breadcrumbs';
export { default as Contacts } from './layout/contacts';
export { default as Cost } from './home/cost';
export { default as FAQItem } from './FAQ/fag-item';
export { default as FAQList } from './FAQ/fag-list';
export { default as Feedback } from './home/feedback';
export { default as FooterMenu } from './layout/footer-menu';
export { default as GridItem } from './home/grid-item';
export { default as Hero } from './home/hero';
export { default as Logo } from './layout/logo';
export { default as Menu } from './layout/menu';
export { default as Performers } from './home/performers';
export { default as PriceCalculator } from './product-price-calculator';
export { default as Promotions } from './home/promotions';
export { default as ScrollController } from './layout/scroll-controller';
export { default as Services } from './home/services';
export { default as ThemeSwitcher } from './layout/theme-switcher';
export { default as ToggleMenuTrigger } from './layout/toggle-controls';
export { default as Workflow } from './performers/collaboration-workflow';
