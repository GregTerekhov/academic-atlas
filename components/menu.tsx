import { MobileMenuTemplate } from 'template';

export default function Menu() {
  return (
    <>
      <MobileMenuTemplate>
        <nav>Menu</nav>
      </MobileMenuTemplate>
      <nav className='hidden lg:flex'>Menu</nav>
    </>
  );
}
