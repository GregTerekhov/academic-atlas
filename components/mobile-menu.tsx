export default function MobileMenu() {
  return (
    <>
      <p>MobileMenu</p>
    </>
  );
}
