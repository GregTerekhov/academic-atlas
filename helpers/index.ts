export * from './anchorLinks';
export * from './breadcrumbsHelper';
export * from './calculatePrice';
export * from './calculationHelper';
export * from './cookiesHelper';
export * from './componentsHelper';
export * from './layoutsHelper';
export * from './mapArray';
export * from './swiperConfig';
export * from './tailwindPlugins';
export * from './telegramHelper';
export * from './templateHelper';
export * from './themeHelper';
export * from './toggleScrollLock';
export * from './validateInput';
