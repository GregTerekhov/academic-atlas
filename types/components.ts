import { AriaLabel } from './aria';
import { WorkType } from './calculation';
import { CompanyContacts } from './layoutTypes';
import { IconName } from './ui';

export enum ThemeVariants {
  DARK = 'dark',
  LIGHT = 'light',
}

export enum StatisticCount {
  Year = '15+',
  Service = '40+',
  Speciality = '50+',
  Work = '2850+',
  Expert = '25+',
}

export enum StatisticLabel {
  Year = 'років діяльності',
  Service = 'видів послуг',
  Speciality = 'спеціальностей',
  Work = 'якісно виконаних робіт',
  Expert = 'експертна команда',
}

export enum BenefitLabel {
  Uniqueness = 'Високий рівень унікальності',
  Guarantee = 'Гарантія якості та результатів',
  Correction = 'Можливість внесення правок',
  Support = 'Підтримка до захисту',
}

export enum AriaLabelTrigger {
  Default = 'Кнопка відкриття навігаційного бургер-меню',
  CloseNavigation = 'Кнопка закриття навігаційного бургер-меню',
  CloseCalculation = 'Кнопка закриття меню калькуляції',
}

export interface IBaseId {
  id: string;
}

interface IBaseTitleDescription {
  title: string;
  description: string;
}

interface IBaseIcon {
  iconName: IconName;
}

interface IBaseImageData {
  imageSrc: string;
  imageAlt: string;
}

export interface IServiceItem extends IBaseId, IBaseImageData {
  serviceTitle: WorkType;
  priority?: boolean | undefined;
}

export interface IBenefitsItem extends IBaseId, IBaseIcon {
  label: BenefitLabel;
}

export interface ISlide extends IBaseId, IBaseTitleDescription, IBaseImageData {
  memberRating: number;
}

export interface IOrderStep extends IBaseId, IBaseIcon {
  step: string;
}

export interface IAboutUs extends IBaseId, IBaseTitleDescription, IBaseImageData {}
export interface IPartnershipBenefits extends IBaseId, IBaseIcon, IBaseTitleDescription {}
export interface IRequirements extends IBaseId, IBaseTitleDescription {}

export interface IWorkflow extends IBaseId, IBaseTitleDescription {
  gridMarkup: string;
}

export interface IAccession extends IBaseId {
  desc: string;
}

export interface IHeroGrid extends IBaseId {
  className: string;
  imageSrc?: string;
  imageAlt?: string;
}

export interface ISubItem {
  id: string;
  textField: string;
}

export interface ILegalInfoArticle {
  id: number;
  article: string;
  paragraph: Record<string, string | { title: string; subItems: ISubItem[] }>;
}

export interface ISubstituteProps {
  value: string;
  substitute: CompanyContacts | string;
  ariaLabel: AriaLabel;
  isInternalLink: boolean;
}

export interface IRequisites {
  id: string;
  fieldName: string;
}

export interface IStatisticItem {
  id: string;
  count: StatisticCount;
  label: StatisticLabel;
  hideOnSmallScreen?: boolean;
  hideOnLargeScreen?: boolean;
  showOnLargeScreen?: boolean;
}
