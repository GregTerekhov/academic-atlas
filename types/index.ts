export * from './aria';
export * from './calculation';
export * from './components';
export * from './FAQ';
export * from './layoutTypes';
export * from './template';
export * from './ui';
