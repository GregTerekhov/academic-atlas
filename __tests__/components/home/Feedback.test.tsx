import { render, screen } from '@testing-library/react';

import { BACKGROUNDS, SectionDescriptions, SectionTitle } from 'types';
import { DEVICES, mapArray } from 'helpers';

import { Feedback } from 'components';

jest.mock('template', () => ({
  SectionTemplate: jest.fn(({ children }) => (
    <section
      data-testid='section-feedback'
      id='feedback'
    >
      <h2>{SectionDescriptions[SectionTitle.CustomerReviews]}</h2>
      {children}
    </section>
  )),
}));

jest.mock('ui', () => ({
  CarouselUI: jest.fn(({ slides }) => (
    <div data-testid='carousel-ui'>
      {slides.map(({ ...props }) => (
        <div key={props.id}>
          <p>{props.title}</p>
          <p>{props.description}</p>
          <div>{props.memberRating}</div>
        </div>
      ))}
    </div>
  )),
}));

jest.mock('data', () => ({
  getFeedbackSlides: jest.fn(() => [
    {
      id: '1',
      title: 'Test Slide 1',
      imageSrc: '/images/customer-01.jpg',
      imageAlt: 'First image',
      description: 'Test description 1',
      memberRating: 5,
    },
    {
      id: '2',
      title: 'Test Slide 2',
      imageSrc: '/images/customer-02.jpg',
      imageAlt: 'Second image',
      description: 'Test description 2',
      memberRating: 4,
    },
  ]),
  getSectionProps: jest.fn(() => ({
    homeFeedback: { title: SectionTitle.CustomerReviews, id: 'test-feedback-id' },
  })),
}));

jest.mock('helpers', () => ({
  mapArray: jest.fn(),
  getIdValues: jest.fn(() => ({ Feedback: SectionTitle.CustomerReviews })),
  generateBackgroundImagePaths: jest.fn((section: SectionTitle) => {
    const baseName = BACKGROUNDS[section];
    if (!baseName) return null;

    return DEVICES.reduce(
      (paths, device) => {
        paths[device] = `/backgroundImage/${baseName}-${device}.webp`;
        return paths;
      },
      {} as Record<string, string>,
    );
  }),
}));

describe('Feedback Component', () => {
  beforeEach(() => {
    jest.clearAllMocks();

    (mapArray as jest.Mock).mockImplementation((array, callback) => array.map(callback));
  });

  it('should render Feedback component correctly', () => {
    const { container } = render(<Feedback />);

    const slides = screen.getAllByTestId('carousel-ui')[0].children;
    expect(slides).toHaveLength(2);
    expect(slides[0]).toHaveTextContent('Test Slide 1');
    expect(slides[0]).toHaveTextContent('Test description');
    expect(slides[0]).toHaveTextContent('5');

    expect(slides[1]).toHaveTextContent('Test Slide 2');
    expect(slides[1]).toHaveTextContent('Test description');
    expect(slides[1]).toHaveTextContent('4');

    const sectionElement = screen.getByTestId('section-feedback');
    expect(sectionElement).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('should pass the correct props to SectionTemplate and render the correct h2', () => {
    render(<Feedback />);

    const heading = screen.getByRole('heading', {
      level: 2,
      name: SectionDescriptions[SectionTitle.CustomerReviews],
    });
    expect(heading).toBeInTheDocument();

    const sectionElement = heading.closest('section');
    expect(sectionElement).toHaveAttribute('id', 'feedback');
  });
});
