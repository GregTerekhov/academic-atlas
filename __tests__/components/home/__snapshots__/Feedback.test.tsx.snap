// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback Component should render Feedback component correctly 1`] = `
<div>
  <section
    data-testid="section-feedback"
    id="feedback"
  >
    <h2>
      Відгуки наших клієнтів
    </h2>
    <div
      data-testid="carousel-ui"
    >
      <div>
        <p>
          Test Slide 1
        </p>
        <p>
          Test description 1
        </p>
        <div>
          5
        </div>
      </div>
      <div>
        <p>
          Test Slide 2
        </p>
        <p>
          Test description 2
        </p>
        <div>
          4
        </div>
      </div>
    </div>
  </section>
</div>
`;
